### DISCUSSION TITLE
Realtime Channel Authentication

### DISCUSSION THREAD

**Discussion by peterkogo** [supabase/supabase#10706] [Feature Requests]
### Update: [Realtime Channel Authentication is in progress](https://github.com/orgs/supabase/discussions/22484)

---

Right now it is possible to listen and broadcast to any realtime channel (apart from Postgres CDC) without authentication. This invites malicious intent and is insecure.
This makes it impossible to use supabase for any non-public applications which are based on private workspaces.
Relevant discussion: #8162

**Comment by BeMoreDog** [comment]
I too would love to see this implemented.

**Comment by Rubenburdin** [comment]
This shipped. Channels now support RLS authorization.
Set private: true on your channel and define policies on realtime.messages:

```
create policy "workspace_access"
on realtime.messages
for all
using (
  realtime.topic() = 'workspace:' || auth.jwt()->>'workspace_id'
);
```

```
const channel = supabase.channel('workspace:123', {
  config: { private: true }
})

```
Docs here (https://supabase.com/docs/guides/realtime/authorization)
